### PR TITLE
Add support for custom newsletters

### PR DESCRIPTION
**Pull request by alanmoo**
For #691 

This allows (permissioned) users to type in the newsletter name when creating a signup component, and passes that name to the signup form when it gets used on opportunity pages.

### CODE REVIEW

File: network-api/app/networkapi/landingpage/models.py [hunk 1]
```diff
@@ -54,4 +60,4 @@ def __str__(self):
 
     class Meta:
         verbose_name = 'Custom Page'
-        verbose_name_plural = 'Custom Pages'
+        verbose_name_plural = 'Custom Pages'
```
Comment (gideonthomas) [inline network-api/app/networkapi/landingpage/models.py]: undo this change?

File: network-api/app/networkapi/landingpage/models.py [hunk 2]
```diff
@@ -19,6 +19,12 @@ class Signup(models.Model):
         help_text="Body of signup component"
     )
 
+    newsletter = models.CharField(
+        max_length=100,
+        help_text="The (pre-existing) SalesForce newsletter to sign up for",
+        default="mozilla-leadership-network"
```
Comment (gideonthomas) [inline network-api/app/networkapi/landingpage/models.py]: do you want to allow them to leave the field blank and then in the db it defaults to this? Or is this just a default for existing entries and they always have to put something in for this field in the admin interface?

File: network-api/app/networkapi/templates/pages/landingpage.html [hunk 3]
```diff
@@ -69,7 +69,7 @@ <h1 class="h1-white">{{ page.landingpage.header }}</h1>
             <div class="row">{% if page.landingpage.signup %}
               <div class="cms col-md-6">{{ page.landingpage.content | safe }}</div>
               <div class="col-md-5 offset-md-1">
-                <div class="join-us half-width col" data-cta-header="{{ page.landingpage.signup.header | escape }}" data-cta-description="{{ page.landingpage.signup.description | escape }}"></div>
+                <div class="join-us half-width col" data-cta-header="{{ page.landingpage.signup.header | escape }}" data-cta-description="{{ page.landingpage.signup.description | escape }}" data-newsletter="{{ page.landingpage.signup.newsletter }}"></div>
```
Comment (gideonthomas) [inline network-api/app/networkapi/templates/pages/landingpage.html]: don't know if you want to `escape` this as well?

File: network-api/app/networkapi/landingpage/models.py (hunk 2 above)
Comment (alanmoo) [reply network-api/app/networkapi/landingpage/models.py]: The latter. Also the fact that new entries get pre-populated with this in the admin is helpful too.

File: network-api/app/networkapi/templates/pages/landingpage.html (hunk 3 above)
Comment (alanmoo) [reply network-api/app/networkapi/templates/pages/landingpage.html]: Nope, just a string here, not HTML.

**Comment by alanmoo** [comment]
@gideonthomas I re-added the newline, think I need you to approve again.
